WordPress/src/main/java/org/wordpress/android/util/LiveDataUtils.kt

```diff
@@ -163,6 +163,16 @@ fun <T, U> LiveData<T>.map(mapper: (T) -> U?): MediatorLiveData<U> {
     return result
 }
 
+/**
+ * Calls the specified function [block] with `this` value as its receiver and returns `this` value.
+ */
+fun <T> LiveData<T>.apply(block: LiveData<T>.(T) -> Unit): LiveData<T> {
```

Comment:
what do you think about calling this `doOnNext` as in RxJava? It's basically a sideeffect

Reply:
You're right, `apply()` does imply you are modifying the value. But `doOnNext()` is quite RxJava-specific and to someone who's not familiar with it, it might be confusing in what it actually does. What about `peform()`?

Reply:
`perform` is better than `apply` and I can't think of anything better so 👍 thanks